src/Compiler/SyntaxTree/LexHelpers.fs

```diff
@@ -98,10 +99,8 @@ let mkLexargs
     }
 
 /// Register the lexbuf and call the given function
-let reusingLexbufForParsing lexbuf f =
+let reusingLexbufForParsing (lexbuf: Lexbuf) f =
     use _ = UseBuildPhase BuildPhase.Parse
-    LexbufLocalXmlDocStore.ClearXmlDoc lexbuf
```

Comment:
Where did the ClearXmlDoc calls move please?

Reply:
I removed them, see the last paragraph in the OP above.